optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -541,7 +558,8 @@ public ScoreDefinition buildDeprecatedScoreDefinition() {
                     "constraintProviderClass", this.constraintProviderClass);
             ConfigUtils.applyCustomProperties(constraintProvider, "constraintProviderClass",
                     constraintProviderCustomProperties, "constraintProviderCustomProperties");
-            return new ConstraintStreamScoreDirectorFactory<>(solutionDescriptor, constraintProvider);
+            ConstraintStreamImplType constraintStreamImplType_ = defaultIfNull(constraintStreamImplType, ConstraintStreamImplType.BAVET);
```

Comment:
IMHO Drools should be the default as, in terms of features, it is expected to be more complete sooner.

Reply:
Let's discuss this when Drools surpasses Bavet in features.